File: openmp-fort/parallel-section-example.f90
PROGRAM Parallel_Section_Example

INTEGER :: sum = 0
INTEGER :: i

!$omp parallel sections reduction(+:sum)
!$omp section
  do i = 1,10
    sum = i + sum
  end do
!$omp section
  do i = 1,100
    sum = i + sum
  end do
!$omp section
  do i = 1,1000
    sum = i + sum
  end do
!$omp end parallel sections

PRINT *, "Total sum of workers: ", sum

END PROGRAM Parallel_Section_Example

File: openmp-c/parallel-section-example.c
#include <stdio.h>
#include <omp.h>

int main(int argc, char const *argv[]) {

  int sum = 0;

  #pragma omp parallel sections reduction(+:sum)
  {
    #pragma omp section
    for (int i = 1; i < 11; i++) {
      sum += i;
    }
    #pragma omp section
    for (int i = 1; i < 101; i++) {
      sum += i;
    }
    #pragma omp section
    for (int i = 1; i < 1001; i++) {
      sum += i;
    }
  }

  printf("Total sum of 3 loops: %d\n", sum);

  return 0;
}
